tests/test_frame.py

```diff
@@ -1066,6 +1066,25 @@ def test_types_groupby_as_index() -> None:
     )
 
 
+def test_types_groupby_as_index_list() -> None:
+    """Test type of groupby.size method depending on list of grouper GH1045."""
+    df = pd.DataFrame({"a": [1, 1, 2], "b": [2, 3, 2]})
+    check(
+        assert_type(
+            df.groupby("a", as_index=False).size(),
+            pd.DataFrame,
+        ),
+        pd.DataFrame,
+    )
+    check(
+        assert_type(
+            df.groupby("a", as_index=True).size(),
+            "pd.Series[int]",
+        ),
+        pd.Series,
+    )
+
```

Comment:
can  you add a test that doesn't include the `as_index` argument?

Reply:
Sorry the issue wasn't there, I did a bad copy paste so it was the same test.
Now the tests are different. which includes the test case from the issue.
As well I have added a third case in both tests where we don't specify as_index as you mentioned above.